### PR TITLE
tsp, support "is Record<>"

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
tested locally with 

```ts
model ModelIsRecordUnknown is Record<unknown> {
  @doc("The name property")
  name: string;
}
```

treat is same as `model ModelIsRecordUnknown extends Record<unknown> { name }`

wait cadl-ranch for formal test https://github.com/Azure/cadl-ranch/pull/330

### CODE REVIEW

File: typespec-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -1231,7 +1231,7 @@ export class CodeModelBuilder {
       case "Model":
         if (isArrayModelType(this.program, type)) {
           return this.processArraySchema(type, nameHint);
-        } else if (isRecordModelType(this.program, type)) {
+        } else if (isRecordModelType(this.program, type) && type.properties.size == 0) {
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:1234]: the "pure" Record...

File: typespec-extension/src/code-model-builder.ts [hunk 2]
```diff
@@ -1598,7 +1598,14 @@ export class CodeModelBuilder {
             }
           });
         }
+      } else {
+        pushDistinct(objectSchema.parents.all, parentSchema);
       }
+    } else if (isRecordModelType(this.program, type)) {
```
Comment (haolingdong-msft) [inline typespec-extension/src/code-model-builder.ts:1604]: Do we need pure Record check here?

Comment (weidongxu-microsoft) [reply typespec-extension/src/code-model-builder.ts:1604]: No, if it gets here (the `processObjectSchema`), it is already mixed.

pure one already handled above with `processObjectSchema`

**Comment by weidongxu-microsoft** [comment]
Emm, cadl-ranch get updated. Refreshing.

File: typespec-tests/src/test/java/com/encode/duration/EncodeDurationTests.java [hunk 3]
```diff
@@ -31,6 +33,8 @@ public void testQuery() {
         queryClient.int32Seconds(SECOND36);
 
         queryClient.iso8601(DAY40);
+
+//        queryClient.int32SecondsArray(Arrays.asList(SECOND36, Duration.ofSeconds(47)));
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/encode/duration/EncodeDurationTests.java:37]: there is bug on this one, will fix in a separate PR (as this one is not about duration).

**Pull request merged**